src/store/index.js

```diff
@@ -13,145 +13,166 @@ import remoteConnection from './plugins/remoteConnection'
 import notificationOnRemoteConnection from './plugins/notificationOnRemoteConnection'
 import decryptAccounts from './plugins/decryptAccounts'
 import aeppApi from './plugins/aeppApi'
+import promisifySecureStorage from '../lib/promisifySecureStorage'
 
 Vue.use(Vuex)
 
-const store = new Vuex.Store({
-  strict: process.env.NODE_ENV !== 'production',
-  plugins: [
-    createPersistedState({
-      paths: [
-        'peerKey',
-        ...IS_MOBILE_DEVICE
-          ? [
-            'apps',
-            'rpcUrl',
-            'mobile.keystore',
-            'mobile.accountCount',
-            'selectedIdentityIdx',
-            'addressBook',
-            'mobile.followers',
-            'mobile.names'
-          ] : []
-      ],
-      setState: (key, state, storage) =>
-        storage.setItem(key, JSON.stringify(state, (key, value) =>
-          value instanceof ArrayBuffer
-            ? { type: 'ArrayBuffer', data: Array.from(new Uint8Array(value)) }
-            : value)),
-      getState: (key, storage) =>
-        JSON.parse(storage.getItem(key), (key, value) =>
-          value && value.type === 'ArrayBuffer'
-            ? new Uint8Array(value.data).buffer
-            : value)
-    }),
-    pollBalance,
-    initEpoch,
-    remoteConnection.plugin,
-    aeppApi,
-    ...IS_MOBILE_DEVICE
-      ? [decryptAccounts, notificationOnRemoteConnection] : []
-  ],
+export default function (storage) {
```

Comment:
If `vuex-persist` plugin can't wait for initialization of secure storage, we should write a custom vuex plugin instead of wrapping the whole store. But `vuex-persist` should call `getItem` asynchrony, maybe secure storage can be wrapped in a way that first call of `getItem` should initialize it.

Reply:
> If we add plugins and blackboxes on everything, thigs will go out of control!

No, it is not.

This way of implementation increases technical debt, if you want, I can fix it by myself.

Reply:
You're over-engineering things

https://en.wikipedia.org/wiki/KISS_principle